### PR TITLE
Added template for core dev documentation

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Added initial pages for developer documentation
 - [x] Create tickets for improving documentation

### Tickets:
 - 68938

### CODE REVIEW

File: docs/dev/index.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    openvino--infer--->gpu
+    openvino--infer--->auto
+```
+
+## [Get started](./get_started.md)
+
+ * [Build OpenVINO](./build.md)
+ * [API reference](https://docs.openvino.ai/latest/api/api_reference.html)
+ * [Samples](../../samples/)
+ * [Tutorials](../../README.md#tutorials)
```
[the remark this answers is not in the document]
Comment (ilyachur) [reply docs/dev/index.md]: Fixed

**Review by ilya-lavrenov — APPROVED** [review]

**Review by tsavina — COMMENTED** [review]
just started to make some suggestions to improve writing. I can see that the PR is merged, but still leaving the comments so I don't lose them in the future.  Probably will make a separate PR then.

**Comment by ilyachur** [comment]
@tsavina Thank you for the feedback, I will fix it in the separate PR

**Comment by tsavina** [comment]
@ilyachur sure thing, thanks) I would still like to fix the rest of docs mentioned in this PR. Just assign me in your separate PR or I'll make one when I get to this).

**Comment by ilyachur** [comment]
@tsavina Ok, thank you. Also today I sent email to you with ask for articles from Wiki pages, could you please take a look and answer if you have some resources to help us.

**Comment by tsavina** [comment]
@ilyachur yeah about that, we'll need some time to finish the current release first. We do have resources but we need some time to do this. Let's get back to this topic on Wednesday/Thursday, if that's ok with you.

**Pull request merged**